### PR TITLE
[v1.29] Add k8s v1.29.3 for RKE1, RKE2 and k3s 

### PR DESCRIPTION
**Pull request by krunalhinguu**
### Updates

- Add RKE1, RKE2 and k3s support for k8s v1.29.3
- Add flannel v0.25.1
- Add calico & canal v3.27.3
- Updated `image.go` file to add support for k8s version which requires `minChannelServerVersion` to be greater than or equal to v2.9.0.
- Add k8s v1.29 minor version support in drone provisioning and operations test.

### TODO

- [x] [flannel-cni](https://github.com/rancher/flannel-cni/pull/12) needs to be merged and tagged.

### Related

- https://github.com/rancher/rancher/issues/43110

### CODE REVIEW

File: channels-rke2.yaml
Comment (rancher-max) [inline channels-rke2.yaml]: I think the versions exactly match what is in v1.28.8, so we can just pull from that and not update anything. Ref: https://github.com/rancher/rke2/releases/tag/v1.29.3%2Brke2r1

**Comment by kinarashah** [comment]
@krunalhinguu changes look ok, could you open RKE PR updating KDM data from this PR to verify the build passes before approving/merging? Also, needs a rebase but we can wait on that before final approval

File: channels-rke2.yaml
Comment (rancher-max) [inline channels-rke2.yaml]: Only should have one `featureVersions`

File: channels-rke2.yaml
Comment (rancher-max) [inline channels-rke2.yaml]: can we move this below the 1.28.9 version? We can also include v1.29.4 now too!

File: channels.yaml
Comment (rancher-max) [inline channels.yaml:584]: Let's also include v1.29.4 now too!

**Comment by krunalhinguu** [comment]
updated @rancher-max

**Review by rancher-max — DISMISSED** [review]
LGTM for k3s and rke2, but we could also include v1.29.4 here for them too as those versions have been released and are available in other branches.

**Comment by krunalhinguu** [comment]
I have just removed them, because ci was failing after that. But i am still checking if we can add them once ci passes

**Comment by rancher-max** [comment]
cc @jakefhyde I wonder if CI failing is due to limitations on the rancher front here?

**Review by rancher-max — DISMISSED** [review]
LGTM for k3s and rke2 -- CI failures seem to be indicative of failures with v1.29 provisioning, which will need to be addressed but I don't think is KDM's responsibility itself.

**Comment by krunalhinguu** [comment]
There was some issue with r/r due to which it is failing. Fixed is already there, once it will be merged i will re run and make sure it passes before we merge kdm. @rancher-max

**Review by kinarashah — APPROVED** [review]

**Pull request merged**
